src/main/java/org/lsc/plugins/connectors/james/JamesDao.java

```diff
@@ -273,4 +280,99 @@ public boolean userExists(String user) {
 
 		throw new JamesClientException(usersClient.getUri(), HttpMethod.HEAD, response);
 	}
+
+	public boolean addDomainContact(Contact contact) {
+		Response response = contactClient
+			.path(String.format("/domains/%s/contacts", contact.getDomain()))
+			.request()
+			.header(HttpHeaders.AUTHORIZATION, authorizationBearer)
+			.post(Entity.json(contact));
+
+		String rawResponseBody = response.readEntity(String.class);
+		response.close();
+		if (checkResponse(response)) {
+			LOGGER.debug("Create domain contact {} is successful", contact.getEmail());
+			return true;
+		} else {
+			LOGGER.error(String.format("Error %d (%s - %s) while creating domain contact: %s",
+				response.getStatus(),
+				response.getStatusInfo(),
+				rawResponseBody,
+				contactClient.getUri().toString()));
+			return false;
+		}
+	}
+
+	public List<User> getUsersListViaDomainContacts() {
+		WebTarget target = contactClient.path("/domains/contacts");
+		LOGGER.debug("GETting users with domain contacts list: " + target.getUri().toString());
+		List<String> users = target.request()
+			.header(HttpHeaders.AUTHORIZATION, authorizationBearer)
```

Comment:
can we add an authorization header in the constructor of this class?

Reply:
The theory we can. But .header has to come after .path().request). And here diff cases we have diff paths => difficult to specify the same header.